app/src/main/java/com/example/myapplication/ui/MainViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.myapplication.ui
+
+import androidx.lifecycle.ViewModel
+import com.example.myapplication.model.ApiHandler
+import com.example.myapplication.model.Product
+import com.example.myapplication.model.SimilarProductsInfo
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class MainViewModel : ViewModel() {
+    private val apiHandler = ApiHandler()
```

Comment:
Let's try something :) Modify ApiHandler to be an interface, create a new class e.g: ApiHandlerImpl which is implements this interface and all your function implementation can go there. And here comes the experiment -> modify MainViewModel to have apiHandler: ApiHandler to the constructor. There is a way to get the view model with param from the view model factory. It will be your research task for today :)